### ISSUE TITLE
[Bug] Compiler warning:  reference to free variable `doom-modeline-mode-alist'

### ISSUE THREAD

**Issue by dr-scsi** [seagle0128/doom-modeline#566]
### Thank you for the bug report

- [X] I am using the latest version of `doom-modeline` related packages.
- [X] I checked [FAQ](https://github.com/seagle0128/doom-modeline#faq).
- [X] You may also try reproduce the issue using clean environment and minimal configurations with
the command `emacs -Q`.

### Bug description

When native-compiling the latest version, the compiler issues:

```
In doom-modeline-auto-set-modeline:
doom-modeline.el:160:16: Warning: reference to free variable `doom-modeline-mode-alist'
```

The problem is that the function `doom-modeline-auto-set-modeline`
https://github.com/seagle0128/doom-modeline/blob/31ef377c5317bc433f1559b496286712f550a1e5/doom-modeline.el#L157-L165

 uses `doom-modeline-mode-alist` which is declared further down in the library:
https://github.com/seagle0128/doom-modeline/blob/31ef377c5317bc433f1559b496286712f550a1e5/doom-modeline.el#L178-L194

The warning can be pacified by moving one of the forms around.

### Steps to reproduce

Byte-compile `doom-modeline.el` and let the native-compiler run again.

### Expected behavior

Ideally, no warning is issued.

### OS

Windows

### Emacs Version

29 (gccemacs-dev)

### Emacs Configurations

Vanilla Emacs 29 from git-master

**Comment by seagle0128** [comment]
I believe it has been fixed in 7ce3760.

**Comment by dr-scsi** [comment]
@seagle0128 - Yes, that commit fixes the issue.  Thanks for your prompt action 👍
